pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,658 @@
+                and max wires for the vertical line
+            control_values=None (Union[Bool, Iterable[Bool]]): for each control wire, denotes whether to control
+                on ``False=0`` or ``True=1``.
+            kwargs=None (dict): mpl line keywords
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=3)
+
+            drawer.ctrl(layer=0, wires=0, wires_target=1)
+            drawer.ctrl(layer=1, wires=(0,1), control_values=[0,1])
+            drawer.ctrl(layer=2, wires=(0,1), color="indigo")
+
+        .. figure:: ../../_static/drawer/ctrl.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        wires_ctrl = _to_tuple(wires)
+        wires_target = _to_tuple(wires_target)
+        if control_values is not None:
+            control_values = _to_tuple(control_values)
+
+        wires_all = wires_ctrl + wires_target
+        min_wire = min(wires_all)
+        max_wire = max(wires_all)
+
+        line = plt.Line2D((layer, layer), (min_wire, max_wire), zorder=2, color=color)
+        self.ax.add_line(line)
+
+        if control_values is None:
+            for wire in wires_ctrl:
+                self._ctrl_circ(layer, wire, zorder=3, color=color)
+        else:
+            if len(control_values) != len(wires_ctrl):
+                raise ValueError("`control_values` must be the same length as `wires`")
+            for wire, control_on in zip(wires_ctrl, control_values):
+                if control_on:
+                    self._ctrl_circ(layer, wire, zorder=3, color=color)
+                else:
+                    self._ctrlo_circ(layer, wire, zorder=3, color=color)
+
+    def _ctrl_circ(self, layer, wires, zorder=3, color=None):
+        """Draw a solid circle that indicates control on one"""
+
+        if color is None:
+            kwargs = {"facecolor": plt.rcParams["lines.color"]}
+        else:
+            kwargs = {"color": color}
+
+        circ_ctrl = plt.Circle((layer, wires), radius=self._ctrl_rad, zorder=zorder, **kwargs)
+        self.ax.add_patch(circ_ctrl)
+
+    def _ctrlo_circ(self, layer, wires, zorder=3, color=None):
+        """Draw an open circle that indicates control on zero."""
+
+        kwargs = {
+            "edgecolor": plt.rcParams["lines.color"],
+            "facecolor": plt.rcParams["axes.facecolor"],
+            "linewidth": plt.rcParams["lines.linewidth"],
+        }
+        if color is not None:
+            kwargs["edgecolor"] = color
+
+        circ_ctrlo = plt.Circle((layer, wires), radius=(self._octrl_rad), zorder=zorder, **kwargs)
+
+        self.ax.add_patch(circ_ctrlo)
+
+    def CNOT(self, layer, wires, color=None):
+        """Draws a CNOT gate.
+
+        Args:
+            layer (Int): layer to draw in
+            wires (Int, Int): tuple of (control, target)
+
+        Keyword Args:
+            color (None or str): mpl compatible color designation
```

Comment:
If we'd like to follow the type-hinting convention (as with `Union[...]` instead of separating types with a comma:
```suggestion
            color (Optional[str]): mpl compatible color designation
```
 I think we mix the conventions in PennyLane currently, though. Using both `Union[None, str]`, `None, str`, and `Optional[str]`. I'd say either one of the latter two is preferable, although the final one is the standard type-hinting convention.

Reply:
Actually realizing colors are much more general than just `str`, given they can also be rgb, rgba, or hex. So I'm just going to remove type considerations here.
```suggestion
            color : mpl compatible color designation
```